[qtbase] fix macdeployqt for symlinks

m-kuhn opened the pull request:
When building for `*-osx-dynamic`, symlinks to handle version suffix are created for the shared libs. Macdeployqt does not preserve the symlinks, which results in multiple copies of the same library causing crashes.
See https://bugreports.qt.io/browse/QTBUG-126612

This pull requests fixes macdeployqt to preserve symlinks

Upstream PR https://codereview.qt-project.org/c/qt/qtbase/+/570829

BillyONeal commented:
> Upstream PR https://codereview.qt-project.org/c/qt/qtbase/+/570829

There are unresolved comments in upstream's PR though they have had since November 8 to look at it. I think we normally need to wait until December 8 at a minimum.

m-kuhn commented:
@BillyONeal last conversation was

> Tor Arne
Nov 07
Sorry, wasn't able to get to this. Could you file a bug report that describes the failure case that this is solving? Thanks!
>
> Matthias Kuhn
Nov 07
It's already there: https://bugreports.qt.io/browse/QTBUG-126612
>
> Tor Arne
Nov 07
It wasn't mentioned in the commit message. Please add Fixes: QTBUG-126612
>
> Matthias Kuhn
Nov 08
Thank you for the hint, done

It's probably me not understanding how to mark this resolved

BillyONeal commented:
> @BillyONeal last conversation was
> 
> > Tor Arne
> > Nov 07
> > Sorry, wasn't able to get to this. Could you file a bug report that describes the failure case that this is solving? Thanks!
> > Matthias Kuhn
> > Nov 07
> > It's already there: https://bugreports.qt.io/browse/QTBUG-126612
> > Tor Arne
> > Nov 07
> > It wasn't mentioned in the commit message. Please add Fixes: QTBUG-126612
> > Matthias Kuhn
> > Nov 08
> > Thank you for the hint, done
> 
> ~It's probably me not understanding how to mark this resolved~ Found out how to do it

The problem is that this isn't a statement from upstream that they want this specific change, only that they wanted to see the specific bug the change was intended to fix in order to review it.

I'm not opposed to the change, it looks reasonable to me, but when it comes to "functional source changes" like this we give upstream 30 days.

m-kuhn commented:
Upstream had time since June 24.

Mengna-Li commented:
PR https://github.com/microsoft/vcpkg/pull/42525 is upgrading the Qt version. To avoid conflicts, let's wait for that PR to merge.

Mengna-Li commented:
@m-kuhn  Please fix the conflicts.

m-kuhn committed:
[qtbase] fix macdeployqt for symlinks

when building for *-osx-dynamic, symlinks to handle version suffix are created for the shared libs. Macdeployqt does not preserve the symlinks, which results in multiple copies of the same library causing crashes.
